ui/src/app/shared/components/graph/graph-panel.tsx

```diff
@@ -72,13 +72,23 @@ export const GraphPanel = (props: Props) => {
 
     const visible = (id: Node) => {
         const label = props.graph.nodes.get(id);
-        return (
-            (nodeGenres[label.genre] &&
-                (!nodeClassNames || Object.entries(nodeClassNames).find(([className, checked]) => checked && (label.classNames || '').split(' ').includes(className))) &&
-                (!nodeTags || Object.entries(nodeTags).find(([tag, checked]) => !label.tags || (checked && label.tags.has(tag)))) &&
-                !nodeSearchKeyword) ||
-            label.label.includes(nodeSearchKeyword)
-        );
+        // If the node matches the search string, return without considering filters
+        if (nodeSearchKeyword && label.label.includes(nodeSearchKeyword)) {
+            return true;
+        }
+        // If the node doesn't match enabled genres, don't display
+        if (!nodeGenres[label.genre]) {
```

Comment:
should we add some tests?

Reply:
I spent some time trying, but it's a pain: will need to extract the function out of the component, create parameters for `props` and all the required variables, and then mock them in the test. I'd be happy to do so, but don't think it's worth it. Thoughts?

Reply:
sounds like poor roi, leave it

Reply:
@alexec Ready for review then